frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -385,7 +434,27 @@ function DataFrame({
 
   const isDynamicAndEditable =
     !isEmptyTable && element.editingMode === DYNAMIC && !disabled
+
+  const isRowSelectionActivated =
+    !isEmptyTable &&
+    !disabled &&
+    (element.selectionMode.includes(ArrowProto.SelectionMode.MULTI_ROW) ||
+      element.selectionMode.includes(ArrowProto.SelectionMode.SINGLE_ROW))
+  const isMultiRowSelectionActivated =
+    isRowSelectionActivated &&
+    element.selectionMode.includes(ArrowProto.SelectionMode.MULTI_ROW)
+
+  const isColumnSelectionActivated =
+    !isEmptyTable &&
+    !disabled &&
+    (element.selectionMode.includes(ArrowProto.SelectionMode.SINGLE_COLUMN) ||
+      element.selectionMode.includes(ArrowProto.SelectionMode.MULTI_COLUMN))
+  const isMultiColumnSelectionActivated =
+    isColumnSelectionActivated &&
+    element.selectionMode.includes(ArrowProto.SelectionMode.MULTI_COLUMN)
```

Comment:
suggestion: I personally would move this into a convenience function outside of the component-function

```typescript
function getSelectionModes(isEmptyTable: boolean, isDisabled: boolean, element): [boolean, boolean, boolean, boolean] {
  const isSelectionActivated = !isEmptyTable && !disabled;

  if (!isSelectionActivated) {
    return;
  }

    const isMultiRowSelectionActivated = element.selectionMode.includes(ArrowProto.SelectionMode.MULTI_ROW);
  const isRowSelectionActivated = isMultiRowSelectionActivated ||
      element.selectionMode.includes(ArrowProto.SelectionMode.SINGLE_ROW);
  
  const isMultiColumnSelectionActivated = element.selectionMode.includes(ArrowProto.SelectionMode.MULTI_COLUMN);
  const isColumnSelectionActivated = isMultiColumnSelectionActivated || element.selectionMode.includes(ArrowProto.SelectionMode.SINGLE_COLUMN);

  return [isRowSelectionActivated, isMultiRowSelectionActivated, isColumnSelectionActivated, isMultiColumnSelectionActivated];
}
```

and then in the component it becomes a one-liner:

```typescript
const [isRowSelectionActivated, isMultiRowSelectionActivated, isColumnSelectionActivated, isMultiColumnSelectionActivated] = getSelectionModes(isEmptyTable, isDisabled, element);
```

Reply:
Also if this is in a separate function, it is unit-testable 🙂

Reply:
I moved this out into a dedicated custom hook